examples/experimental/dagster-airlift/dagster_airlift/in_airflow/mark_as_migrating.py

```diff
@@ -0,0 +1,105 @@
+        for task_id in migration_state_for_dag.tasks.keys():
+            if task_id not in dag.task_dict:
+                raise Exception(
+                    f"Task with id `{task_id}` not found in dag `{dag.dag_id}`. Found tasks: {list(dag.task_dict.keys())}"
+                )
+            if not isinstance(dag.task_dict[task_id], BaseOperator):
+                raise Exception(
+                    f"Task with id `{task_id}` in dag `{dag.dag_id}` is not an instance of BaseOperator. This likely means a MappedOperator was attempted, which is not yet supported by airlift."
+                )
+        migrating_dags.add(var)
+
+    for var in migrating_dags:
+        dag: DAG = global_vars[var]
+        logger.debug(f"Tagging dag {dag.dag_id} as migrating.")
+        dag.tags.append(
+            json.dumps(
+                {"DAGSTER_MIGRATION_STATUS": migration_state.get_migration_dict_for_dag(dag.dag_id)}
+            )
+        )
+        migration_state_for_dag = migration_state.dags[dag.dag_id]
+        migrated_tasks = set()
+        for task_id, task_state in migration_state_for_dag.tasks.items():
+            if not task_state.migrated:
+                logger.debug(
+                    f"Task {task_id} in dag {dag.dag_id} has `migrated` set to False. Skipping..."
+                )
+                continue
+
+            # At this point, we should be assured that the task exists within the task_dict of the dag, and is a BaseOperator.
+            original_op: BaseOperator = dag.task_dict[task_id]  # type: ignore  # we already confirmed this is BaseOperator
+            logger.debug(
+                f"Creating new operator for task {original_op.task_id} in dag {original_op.dag_id}"
+            )
+            new_op = build_dagster_task(
+                task_id=original_op.task_id,
+                owner=original_op.owner,
+                email=original_op.email,
+                email_on_retry=original_op.email_on_retry,
+                email_on_failure=original_op.email_on_failure,
+                retries=original_op.retries,
+                retry_delay=original_op.retry_delay,
+                retry_exponential_backoff=original_op.retry_exponential_backoff,
+                max_retry_delay=original_op.max_retry_delay,
+                start_date=original_op.start_date,
+                end_date=original_op.end_date,
+                depends_on_past=original_op.depends_on_past,
+                wait_for_downstream=original_op.wait_for_downstream,
+                params=original_op.params,
+                doc_md="This task has been migrated to dagster.",
+            )
+            original_op.dag.task_dict[original_op.task_id] = new_op
+
+            new_op.upstream_task_ids = original_op.upstream_task_ids
+            new_op.downstream_task_ids = original_op.downstream_task_ids
+            new_op.dag = original_op.dag
+            original_op.dag = None
+            migrated_tasks.add(task_id)
```

Comment:
a help function called `swap_tasks` or something written generically with comments would make this a lot clearer I think. You can cleanly separate the creation of the dagster task from the mutation to the dag that way.